src/transformers/models/swin/modeling_tf_swin.py

```diff
@@ -659,64 +659,46 @@ def __init__(
         self.drop_path = (
             TFSwinDropPath(config.drop_path_rate, name="drop_path")
             if config.drop_path_rate > 0.0
-            else tf.identity(name="drop_path")
+            else tf.keras.layers.Activation("linear", name="drop_path")
         )
         self.layernorm_after = tf.keras.layers.LayerNormalization(
             epsilon=config.layer_norm_eps, name="layernorm_after"
         )
         self.intermediate = TFSwinIntermediate(config, dim, name="intermediate")
         self.swin_output = TFSwinOutput(config, dim, name="output")
 
-    def set_shift_and_window_size(self, input_resolution: Tuple[int, int]) -> None:
```

Comment:
Conditionally resetting the instance variables in both the `__init__` and `call` methods was not compatible with the model in graph mode. Instead `self.shift_size` and `self.window_size` are set in `__init__` with the same logic as this method.

In the `call` method, the local variables `shift_size` and `window_size` are set according to the same logic as this method. **Technically this isn't exactly equivalent** as the condition is dependant on `self.window_size` and in the original logic, `self.window_size` could change after every model call. As our model, at prediction time, should produce the same outputs if called twice on the same inputs, I think the refactor is consistent with expected behaviour.